healthcheck/health.go

```diff
@@ -15,10 +15,10 @@ import (
 type HealthChecker interface {
 	CheckStatus() HealthResponse
 	AddHealthChecker(name string, checker BasicChecker)
-	AddHTTPEndpoint(name string, targetURL string, timeoutDuration time.Duration, expectedStatus int, cacheDuration time.Duration)
-	AddHTTPEndpoints(endpoints map[string]string, timeoutDuration time.Duration, expectedStatus int, cacheDuration time.Duration)
-	AddDatabase(name string, db HealthDatabase, cacheDuration time.Duration)
-	AddAuditEventsReporterModule(name string, reporter events.AuditEventsReporterModule, timeout time.Duration, cacheDuration time.Duration)
+	AddHTTPEndpoint(name string, targetURL string, timeoutDuration time.Duration, expectedStatus int, cacheDuration time.Duration, timeProvider TimeProvider)
```

Comment:
I am wondering if we really need to change the interface.

Could we not in our implementation always set it to our default implementation and just for TestHealthStatusCache build a heathstatus with the mocked version